lib/puma/cluster.rb

```diff
@@ -108,25 +108,32 @@ def spawn_worker(idx, master)
     def cull_workers
       diff = @workers.size - @options[:workers]
       return if diff < 1
+      debug "Culling #{diff} workers"
 
-      debug "Culling #{diff.inspect} workers"
+      workers = workers_to_cull(diff)
+      debug "Workers to cull: #{workers.inspect}"
 
-      workers_to_cull =
-        case @options[:worker_culling_strategy]
-        when :youngest
-          @workers.sort_by(&:started_at)[-diff,diff]
-        when :oldest
-          @workers.sort_by(&:started_at)[0,diff]
-        end
-
-      debug "Workers to cull: #{workers_to_cull.inspect}"
-
-      workers_to_cull.each do |worker|
+      workers.each do |worker|
         log "- Worker #{worker.index} (PID: #{worker.pid}) terminating"
         worker.term
       end
     end
 
+    def workers_to_cull(diff)
+      workers = @workers.sort_by(&:started_at)
+      workers.reject! { |w| w.index == 0 } if @options[:fork_worker]
```

Comment:
Let's leave a comment here saying the same thing you did - that in fork_worker mode, worker 0 is our "master" and culling it would impact CoW gains

Reply:
Done